.github/workflows/nightly.yaml

```diff
@@ -0,0 +1,89 @@
+name: Nightly build
+
+on:
+  schedule:
+    - cron: "0 3 * * *"
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    name: Build integration for
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        goos: [ linux ]
+        goarch: [ amd64, arm64, arm ]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.17'
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - name: Build integration
+        env:
+          GOOS: ${{ matrix.goos }}
+          GOARCH: ${{ matrix.goarch }}
+        run: |
+          make compile
+      - name: Upload artifact for docker build step
+        uses: actions/upload-artifact@v2
+        with:
+          retention-days: 1
+          name: nri-prometheus-${{ matrix.goos }}-${{ matrix.goarch }}
+          path: bin/nri-prometheus-${{ matrix.goos }}-${{ matrix.goarch }}
+  docker:
+    name: Build and push docker images
+    needs: [ build ]
+    runs-on: ubuntu-latest
```

Comment:
This uses `docker buildx` while the real release workflow uses goreleaser: https://github.com/newrelic/nri-prometheus/blob/main/.github/workflows/release-integration.yml#L74=

I think both workflows should be as similar as possible to avoid having one working while the other does not.

Reply:
Sure, going to update it. Now that you mention it... Should we try to unify across all repos as much as possible the release workflow?

Reply:
We should! But `nri-prometheus` is special, because we need both a multiarch docker image, but also multiple binaries per arch, as you can run it on-host.

Using goreleaser to also produce docker images was easier than making the usual docker pipeline also produce binaries. I don't think the result is ideal (complexity is substantially higher), but given that we'll likely get rid of this solution soon... I'm not sure it makes sense to put more time on refactoring the pipeline :)